### ISSUE TITLE
'uses' within a ResourceType. Bug?

### ISSUE THREAD

**Issue by cazala** [mulesoft/api-workbench#95]
Hey,

So, lets say I have a `Library`, like the following one:

```raml
#%RAML 1.0 Library
# this is saved as myLibrary.raml

traits:
  auth:
    headers:
      Authorization:
        pattern: ^Bearer (.*)$
```

And I use that library in one of my `ResourceType`'s:

```raml
#%RAML 1.0 ResourceType
# this is saved as myResource.raml

uses:
  security: !include myLibrary.raml

is: [security.auth]
```

Now, if I want to use that `ResourceType` from my `api.raml` I get an error if I don't include `myLibrary.raml` again, even tho it is already included in `myResource.raml`, like:

![uses-bug](https://cloud.githubusercontent.com/assets/2781777/12209288/b211c8aa-b630-11e5-9d93-2129d7d1cc75.gif)

My API structure looks like this:

<img width="177" alt="api" src="https://cloud.githubusercontent.com/assets/2781777/12209410/6bb51a1e-b631-11e5-9ee8-4432e29080bf.png">

**Comment by ddenisenko** [comment]
Thanks for reporting this, we'll investigate this.

**Comment by trevorwh** [comment]
+1, we are also noticing the same thing, we are needing to include dependent libraries that are within resourceTypes in api.raml

**Comment by rchang-ck** [comment]
+1 seeing this as well

**Comment by usarid** [comment]
Thanks, definitely looks like an important bug, @ddenisenko or @sichvoge will respond here soon.

**Comment by ddenisenko** [comment]
Well, this ticket was reproduced (as I already stated in my first comment) and added to the nearest milestone.

**Comment by ddenisenko** [comment]
Fixed in 0.8.14.

**Comment by cazala** [comment]
thanks!
